lib/rubocop/cop/style/documentation_method.rb

```diff
@@ -0,0 +1,89 @@
+# encoding: utf-8
+# frozen_string_literal: true
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for missing documentation comment for public method.
+      class DocumentationMethod < Cop
+        include AnnotationComment
+        include OnMethodDef
+        MSG = 'Missing top-level %s documentation method comment.'.freeze
+
+        def_node_matcher :constant_definition?, '{def casgn}'
+
+        def on_def(node)
+          _name, body = *node
+
+          line = node.loc.line
+          return if node.ancestors.first.to_a.include? method_type
+          return if (processed_source[0..line].map(&:strip) &
+            %w(private protected)).any?
+          return if namespace?(body)
+          ast_with_comments = processed_source.ast_with_comments
+          return if associated_comment?(node, ast_with_comments)
+          add_offense(node, :keyword, format(MSG, :module))
+        end
+
+        def on_method_def(node, _method_name, _args, _body)
+          _name, _body = *node
```

Comment:
This method is pretty much the same as the one above. You should extract the shared code somewhere.